### ISSUE TITLE
How to organize "child" theme?

### ISSUE THREAD

**Issue by arnisjuraga** [synapticism/wordpress-gulp-starter-kit#58]
I have "parabola" theme, and want to create a child theme, using starter kit.

1. replaced _config.scss values
2. added "template" variable, to include in style.css file

but - if you symlink theme folder from `/wp-content/themes/parabola-child` to whatever I have starter kit folder "build" - Wordpress will not get original parent theme files automatically. And will do it only, if I copy compiled files directly to `/wp-content/thenes/parabola-child/`

**Comment by arnisjuraga** [comment]
`wp_enqueue_script('voidx-header', get_stylesheet_directory_uri() . '/js/' . $file . '.js', $deps = array(
		'jquery' ), filemtime(get_template_directory() . '/js/' . $file . '.js'), false);`

and

`wp_enqueue_script($name, get_stylesheet_directory_uri() . '/js/' . $file . '.js', array(
		'jquery' ), filemtime(get_template_directory() . '/js/' . $file . '.js'), true);` 

will give an error, because `get_template_directory()` returns Parent theme folder.

`

**Comment by synapticism** [comment]
This sounds like something I'll have to look into. I'm going to be working on a pretty big update to this project soon and will take this into account.

**Comment by mware** [comment]
I'm stuck just trying to figure out where to place the starter kit. Do all of the files go in the theme or in the root directory? And how do you get/point to the final style.css and functions.php out of the build folder and into the theme root folder?
Thx.

**Comment by noeljackson** [comment]
@mware The theme goes in src folder. This is your work directory for JUST the theme files.

When you run gulp it builds from src to build. Then if you run gulp dist, you get a fully built production ready theme inside of the dist folder.

**Comment by mware** [comment]
Thanks @noeljackson.  I have a local dev environment and my issue is that WordPress can't find the build files. Where does the src folder go in relation to the root? If the theme is in the src folder and is then outputted to build, how maintain the typical WordPress structure for the wp-content directory?

Is there an example of how you would structure the entire WordPress install from the root down? 

Something along the lines of 

```
|- root
	|- wp-admin
	|- wp-includes
	|- wp-content
		|--themes
			??
```

**Comment by synapticism** [comment]
@mware I think what you're looking for is symlinking. Your theme development and local copy of WP can be totally separate, just symlink the `build` directory to your `themes` folder. Read the docs for more about that.

**Comment by mware** [comment]
Thx @synapticism. Sorry for being so thick.

**Comment by arnisjuraga** [comment]
I have set it up, that:
~~~
|- root
	|- wp-admin
	|- wp-includes
	|- wp-content
		|--themes
			MyTheme
			wp-gulp      <-- src goes here. And config 
                                             files are modified to build dist files into ../MyTheme
                                            "MyTheme" is defined as variable.
~~~


In the result - I can easily build theme,
I can set up 2 .git repos - one for WP itself, and another for theme source.

**Comment by isaac-martin** [comment]
Question on this - How can I have a git repo in the dist? The .git doesn't get built.

**Comment by arnisjuraga** [comment]
Why not create git repo one folder above? Like in `../dist` ?

**Comment by isaac-martin** [comment]
Yeah my mistake - all sorted! For anyone that stumbles across this i ended up using `gulp-deploy-git' and building it into the dist task.
